apps/gallery/test/marionette/load_valid_image_test.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+var Gallery = require('./lib/gallery.js'),
+	//ImageView = require('./lib/fullscreen_image.js'),
+    Marionette = require('marionette-client'),
+    assert = require('assert');
+
+marionette('loading images', function() {
+
+  var app, fullscreen_view, actions, client;
+
+  client = marionette.client({
+    profile: {
+      prefs: {
+        'device.storage.enabled': true,
+        'device.storage.testing': true,
+        'device.storage.prompt.testing': true
+      }
+    }
+  });
+
+  setup(function() {
+    // Remove all files in temp device storage.
+    client.fileManager.removeAllFiles();
+    // Add file into the pictures directory
+    client.fileManager.add([
+    	{type: 'pictures',filePath: 'apps/gallery/test/images/01.jpg'}, 
+    	{type: 'pictures',filePath: 'apps/gallery/test/images/02.png'}, 
+    	{type: 'pictures',filePath: 'apps/gallery/test/images/03.gif'}, 
+    	{type: 'pictures',filePath: 'apps/gallery/test/images/04.bmp'}, 
+    	// Progressive JPG 
+        {type: 'pictures',filePath: 'apps/gallery/test/images/05.jpg'}, 
+    	// Transparent background PNG
+        {type: 'pictures',filePath: 'apps/gallery/test/images/06.png'}, 
+    	// Animated GIF 
+        {type: 'pictures',filePath: 'apps/gallery/test/images/07.gif'}, 
+    	// Animaged PNG 
+        {type: 'pictures',filePath: 'apps/gallery/test/images/08.png'}  
+    ]);
+    app = new Gallery(client);
+    actions = new Marionette.Actions(client);
+    app.launch();
+  });
+
+  test('open valid images in fullscreen view', function() {
+
+    for (var i = 0; i < 8; i++) {
+    	fullscreen_view = app.tapThumbnail(i);
+        assert.ok(fullscreen_view.displayed);
```

Comment:
I made fullscreen_view.displayed() and app.thumbnailsViewDisplayed() getter methods, because putting waitfor in here won't be allowed since I'm in the loop.  Both methods wait until the view is present, and check for display status.